src/app/auth.tsx

```diff
@@ -0,0 +1,66 @@
+"use client";
+
+import React, { useEffect } from "react";
+import { usePathname, useRouter } from "next/navigation";
+import { createClientComponentClient } from "@supabase/auth-helpers-nextjs";
+import { DatabaseAutoType } from "@/supabase";
+
+export const AuthRouting: React.FC<{ children: React.ReactNode }> = ({ children = <></> }) => {
+    const supabase = createClientComponentClient<DatabaseAutoType>();
+    const router = useRouter();
+    const pathname = usePathname();
+    const [loggedIn, setLoggedIn] = React.useState<boolean>();
+
+    // This fixes issues with navigation when the user has signed in but then signed out and caching still allows them to
+    // access pages. It also fixes authentication flow issues on sign in and sign out
+    const toRedirectTo = (): string | null => {
+        if (loggedIn === undefined) {
+            return null;
+        }
+
+        if (loggedIn) {
```

Comment:
can these be combined to `if (loggedIn && ...) return; if (!loggedIn && ...) return;`? I hate else